src/main/java/com/zaxxer/hikari/pool/PoolElf.java

```diff
@@ -281,15 +284,6 @@ void resetConnectionState(final PoolBagEntry poolEntry) throws SQLException
       }
    }
 
-   void resetPoolEntry(final PoolBagEntry poolEntry)
-   {
-      poolEntry.setReadOnly(isReadOnly);
-      poolEntry.setCatalog(catalog);
-      poolEntry.setAutoCommit(isAutoCommit);
-      poolEntry.setNetworkTimeout(networkTimeout);
-      poolEntry.setTransactionIsolation(transactionIsolation);
-   }
-
```

Comment:
All that for the purpose of removing ``resetPoolEntry()``?

This is a lot of change, that makes the code harder to follow, move responsibilities to unobvious places, and is ultimately in such a low traffic path  it likely gets executing less than once per minute.